source/initial_composition/slab_model.cc

```diff
@@ -0,0 +1,137 @@
+    }
+
+
+    template <int dim>
+    void
+    SlabModel<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Initial composition model");
+      {
+        Utilities::AsciiDataBoundary<dim>::declare_parameters(prm,
+                                                              "$ASPECT_SOURCE_DIR/data/initial-composition/slab-model/",
+                                                              "shell_3d.txt",
+                                                              "Slab model",
+                                                              /*time dependent = */ false);
+      }
+      prm.leave_subsection();
+    }
+
+
+    template <int dim>
+    void
+    SlabModel<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Initial composition model");
+      {
+        slab_boundary.initialize_simulator (this->get_simulator());
+        slab_boundary.parse_parameters(prm, "Slab model", /*time dependent = */ false);
+      }
+      prm.leave_subsection();
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace InitialComposition
+  {
+    ASPECT_REGISTER_INITIAL_COMPOSITION_MODEL(SlabModel,
+                                              "slab model",
+                                              "An initial composition model that implements subducted "
+                                              "slab geometries as a compositional field determined from "
+                                              "an input file. The file defines the depth to the top of "
+                                              "the slab and the slab thickness. "
+                                              "The computed compositional value is 1 within the slabs "
+                                              "and zero elsewhere. "
+                                              "An example model that is included is Slab2 described in "
+                                              "Hayes, G. P., Moore, G. L., Portner, D. E., Hearne, M., "
+                                              "Flamme, H., Furtney, M., \\& Smoczyk, G. M. (2018). Slab2, "
+                                              "a comprehensive subduction zone geometry model. Science, "
+                                              "362(6410), 58-61. The script to convert the Slab2 model "
+                                              "into an aspect input data file is available in the directory "
+                                              "data/initial-composition/slab-model/. Please note that "
+                                              "Slab2 and the example data file assume spherical geometry "
+                                              "(latitude, longitude coordinates), however, that is not "
+                                              "necessary for this plugin, data files in "
+                                              "cartesian coordinates will work with box geometries.")
```

Comment:
But in the initialization, we have an assert and the model will crash in cartesian coordinates. So at the moment, this does not work in box geometries.

Reply:
I don't think the assert on the geometry model was intended to be there, I will remove it. Thanks!